src/boot/bootctl.c

```diff
@@ -2293,7 +2310,7 @@ static int verb_remove(int argc, char *argv[], void *userdata) {
                 return r;
         }
 
-        char *path = strjoina("/EFI/systemd/systemd-boot", get_efi_arch(), ".efi");
+        char *path = strjoina(arg_loader_path,"systemd-boot", get_efi_arch(), ".efi");
```

Comment:
First, thanks for looking at this... And your right it needs some discussion first, which is why I didn't remove the bashisms yet. It's possible the zboot stuff will land in fedora and be enabled soon, so that part may not even need to be downstream.

So, I can see the want/need to have a single systemd-boot, but I just don't see how it's going to work. Right now few machines have distro enrolled keys, and I can't guess which vendors might enroll which keys on which hardware, and even if MS signs a db update with some portion of the distro keys I suspect there will remain a situation where only a subset of the distro's keys are in place on any given machine. And while systemd-boot may remain fairly consistent between distros' unless a 3rd key just for the bootloader is in the firmware the risk exists if say systemd-boot is signed by fedora, and is being used to boot fedora as well as another distro (signed with a different key) and the fedora key needs to be revoked/whatever that the other distro would expect an alternate boot path to remain. Never mind the situation about which distro gets to own the systemd-boot used to boot them, or the naming. Given the other patch to update the UEFI name means that the UEFI boot selection menus themselves are likely going to read $DISTRO.  It seems a big mess of issues just to avoid having a couple of copies of systemd-boot in different EFI/$DISTRO directories. Multiple copies aren't great, but in my mind it simplifies the boot process, each distro gets its own efi entry, and its own key used to validate the entire boot path. The loaders/entries allow each systemd-boot to possibly boot other distro's from a single menu if the keys are setup right.

So, as I mentioned in the second comment, I don't like the way this set behaves. It works for installing/etc, but I would prefer that if its installed to an alternate location that the tool is a bit sticker about it. So I'm probably going to rework it to have a couple compile time defaults/etc. Or maybe if everyone thinks its a bad idea, then we can try sharing a single loader...

Reply:
I'm not sure I follow the use case. As explained earlier, sd-boot does not embed any key/signature/hash. So if you don't want to take shim+sd-boot from the same vendor (so that the key embedded in shim verifies sd-boot), you can simply ensure that whatever key is used to sign sd-boot is enrolled in MoK. Then which shim is used no longer matters (well, there's the SBAT stuff coming, but that's orthogonal).